ArjunaJTA/jms/src/main/java/org/jboss/narayana/jta/jms/JmsXAResourceRecoveryHelper.java

```diff
@@ -177,4 +168,10 @@ private void disconnect() throws XAException {
         }
     }
 
+    private void assertIfRecoveryScanWasStarted() throws XAException {
+        if (delegate == null) {
+            throw new XAException("Recovery scan has to be started");
```

Comment:
In that case I will create a separate JIRA for this. Because there are other places that need changing outside of JBTM-2638 scope.